### DISCUSSION TITLE
How to display bold chinese on MacOS's kitty?

### DISCUSSION THREAD

**Discussion by rainzm** [kovidgoyal/kitty#7241] [Q&A] [answered]
I want to display bold chinese text on MacOS's kitty and I use fira code for ascii char.

As far as I know, symbol_map is mainly designed to display symbols, so it doesn't deal with bold, and I tried, and it is true.

The chinese font will go fallback (Apple CoreText), and as you can see from the `kitty --debug-font-fallback command`, its font fallback goes to PingFang SC Regular:
<img width="1270" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/76140136-79bd-4012-8354-7666788e3034">
And in the figure, U+4f60 bold was also fallback to PingFang SC Regular, even though the font family PingFang SC does indeed have Medium and Semibold, but they were not correctly fallback. 

<img width="896" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/da9402a2-0ad7-450f-b5fb-8e4520633d4c">

So, does this seem to be a problem with Apple's CoreText? Or is it a problem with how kitty is calling CoreText?

And, given that Apple CoreText is out of control, could kitty provide an additional feature to specify the fallback font?

Or, there are other ways to display bold chinese in kitty?

Thank you very much.

**Comment by kovidgoyal** [comment]
When looking for the fallback face, kitty passes the bold/italic flags
however, IIRC there is no way to get CoreText to respect these. If you
know of one, patches are welcome. Relevant code is in
create_fallback_face() in core_text.m

**Comment by kovidgoyal** [comment]
>From what I remember of CoreText, you would probably need to first use 
CTFontCreateForString() as the existing code does. Then once an
appropriate font is found create a font descriptor from the font and
then use CTFontDescriptorCreateCopyWithSymbolicTraits() setting the
weight and style appropriately. And then create a font from that
descriptor, check that the newly created fonts family is the same as the
original fallback font and if so use it instead of the original.

**Comment by rainzm** [comment]
Hi, i add a log in create_fallback_face() on core_text.m file:
<img width="743" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/5083d17b-3ea3-4abf-b7fb-f3abefd756c4">
And, the kitty log:
<img width="2241" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/f639655e-ff7e-4c63-9701-b0ee8431f95a">
Looks like it found PingFangSC-Semibold. But this font is obviously not bold:
<img width="1317" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/caffbf9b-f3b5-4744-9b7e-93fed6cc8b7f">
And, if I specify bold_font in kitty.conf as PingFang SC Semibold. This display font is bold:
<img width="1215" alt="image" src="https://github.com/kovidgoyal/kitty/assets/22927169/73886bf4-24c9-4e75-b2b6-718e055e2db2">

**Reply by kovidgoyal** [reply, marked as answer]
You dont need to do anything, this is already fixed in master: https://github.com/kovidgoyal/kitty/commit/752fcb64241bc1cc2b40e1b44d6f04a89967dd62

**Reply by rainzm** [reply]
Oh bro, you're amazing! Thank you! Kitty is prefect for me now.
